### DISCUSSION TITLE
How can I trigger HotKey when I get the hotkey ID?

### DISCUSSION THREAD

**Discussion by s594569321** [mlo40/VsPyYt#6] [Q&A]
I only find the method to get HotkeyID in Wiki, however is it possible to use in python? Thank you very much. 
--------------------------------------------------------------------------------------------------------------------------
Sorry for ask without thinking more.
Just Create the new custom function in func.py.

```
async def trigger_hotkey(websocket,hotkeyid):
    payload = {
                    "apiName": "VTubeStudioPublicAPI",
                    "apiVersion": v,
                    "requestID": reqid,
                    "messageType": "HotkeyTriggerRequest",
                    "data": {
                            "hotkeyID": hotkeyid,
                    }
            }
    await websocket.send(json.dumps(payload))
    json_data = await websocket.recv()
    pack = json.loads(json_data)
    return pack    
```

**Comment by mlo40** [comment]
I if you are referring to executing hotkeys then i have to say that is a function that i forgot to add, i am adding it right now though and a couple more too, if you need it right this moment then paste this into func.py
```
async def ExHotkey(websocket,hid,IID):
    payload = {
            "apiName": "VTubeStudioPublicAPI",
            "apiVersion": v,
            "requestID": reqid,
            "messageType": "HotkeyTriggerRequest",
            "data": {
                "hotkeyID": hid,
                "itemInstanceID": IID
        }
    }
    await websocket.send(json.dumps(payload))
    json_data = await websocket.recv()
    pack = json.loads(json_data)
    return pack
```

use
ExHotkey(websocket,hotkeyid,"") 
the "" is optional data idk what but it refers to it as iteminstanceID soo good luck

**Comment by mlo40** [comment]
ah shit you already figured it out

**Reply by s594569321** [reply]
Haha, you add the more complete answer. Thank you very much anyway.
